### PR TITLE
enable to set config file base dir by jvm param or System.setProperty

### PR DESCRIPTION
**Pull request by wuzunqian**
### Ⅰ. Describe what this PR did
enable to set config file base dir by System.setProperty

### Ⅱ. Does this pull request fix one issue?

no

### Ⅲ. Why don't you add test cases (unit test/integration test)? 
simple change

### Ⅳ. Describe how to verify it
1.start app with jvm param: -Dseata.config.basedir=/xxx
2.put config file (like registry.conf) in dir /xxx
3.verify if config file effect

### CODE REVIEW

File: config/seata-config-core/src/main/java/io/seata/config/FileConfiguration.java [hunk 1]
```diff
@@ -77,17 +79,23 @@ public FileConfiguration() {
      * @param name the name
      */
     public FileConfiguration(String name) {
+        String baseDir = System.getProperty(CONFIG_BASEDIR);
         if (null == name) {
             CONFIG = ConfigFactory.load();
-        } else {
+        } else if(baseDir == null || baseDir.length() == 0){
```
Comment (xingfudeshi) [inline config/seata-config-core/src/main/java/io/seata/config/FileConfiguration.java]: StringUtils.isNullOrEmpty

**Comment by xingfudeshi** [comment]
Thanks for your contribution!I have a doubt,are these changes used for configuration isolation in different environments?

**Comment by wuzunqian** [comment]
> Thanks for your contribution!I have a doubt,are these changes used for configuration isolation in different environments?

no,i need to load config file from path not only in jar/war, isolation of different enviroments can not satisfy me.

**Comment by xingfudeshi** [comment]
> > Thanks for your contribution!I have a doubt,are these changes used for configuration isolation in different environments?
> 
> no,i need to load config file from path not only in jar/war, isolation of different enviroments can not satisfy me.

Could you please add some test cases to verify this function?

**Comment by a364176773** [comment]
resolve code conflicts

**Pull request closed without merging**